extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -853,109 +932,110 @@ protected function  isCorporateWiki() {
 
 	/**
 	 * This is how we generate the subtemplate for the advanced search box.
+	 *
 	 * @see    SearchControllerTest::testAdvancedBox
 	 * @throws Exception
 	 */
 	public function advancedBox() {
-		$config = $this->getVal('config', false);
-		if ( ! $config instanceof Wikia\Search\Config ) {
-			throw new Exception("This should not be called outside of self-request context.");
+		$config = $this->getVal( 'config', false );
+		if ( !$config instanceof Wikia\Search\Config ) {
+			throw new Exception( "This should not be called outside of self-request context." );
 		}
 
-		$searchEngine = (new SearchEngine);
+		$searchEngine = ( new SearchEngine );
 
 		$searchableNamespaces = $searchEngine->searchableNamespaces();
 
-		wfRunHooks( 'AdvancedBoxSearchableNamespaces', array( &$searchableNamespaces ) );
+		wfRunHooks( 'AdvancedBoxSearchableNamespaces', [ &$searchableNamespaces ] );
 
-		$this->setVal( 'namespaces', 			$config->getNamespaces() );
-		$this->setVal( 'searchableNamespaces', 	$searchableNamespaces );
+		$this->setVal( 'namespaces', $config->getNamespaces() );
+		$this->setVal( 'searchableNamespaces', $searchableNamespaces );
 	}
 
 	/**
 	 * This is how we generate the search type tabs in the left-hand rail
+	 *
 	 * @see    SearchControllerTest::tabs
 	 * @throws Exception
 	 */
 	public function tabs() {
-		$config = $this->getVal('config', false);
+		$config = $this->getVal( 'config', false );
 
-		if ( ! $config || (! $config instanceOf Wikia\Search\Config ) ) {
-			throw new Exception("This should not be called outside of self-request context.");
+		if ( !$config || ( !$config instanceOf Wikia\Search\Config ) ) {
+			throw new Exception( "This should not be called outside of self-request context." );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)